dsl/interpreter.py

```diff
@@ -0,0 +1,164 @@
+    code = simple_transform(data)
+    new_file_name = input_name[:len(input_name) - len(DSL_EXTENSION)] + '.py'
+    with open(new_file_name, 'w') as f:
+        f.write(code)
+    # close file
+    file.close()
+    return new_file_name
+
+def simple_transform(code='RETURN 0'):
+    python_code = 'import numpy as np\n'
+    code = code.replace('\n', ' \n ')
+    code = code.replace('{', ' {')
+    code = code.replace('}', ' }')
+    tokens = code.split(' ')
+    # print(code)
+    # print(tokens)
+    indent = 0
+    prefix = 0
+    if_flag = 0
+    else_flag = 0
+    for_flag = 0
+    for ind, token in enumerate(tokens):
+        if token == '':
+            continue
+        if token == '\n':
+            python_code += '\n' + ' '*(indent*4)
+            continue
+        if token == '{':
+            indent += 1
+            prefix += 1
+            if if_flag > 0:
+                if_flag -= 1
+                python_code += ' : '
+                continue
+            if else_flag > 0:
+                else_flag -= 1
+                python_code += ' : '
+                continue
+            if for_flag > 0:
+                for_flag -= 1
+                python_code += ' : '
+                continue
+
+            continue
+        if token == '}':
+            indent -= 1
+            prefix -= 1
+            if indent < 0 or prefix < 0:
+                logger.error(f'Wrong curly brackets')
+                return
+            continue
+        if token == 'IF':
+            python_code += 'if '
+            if_flag += 1
+            continue
+        if token == 'ELSE':
+            python_code += 'else'
+            else_flag += 1
+            continue
+        if token == 'FOREACH':
+            python_code += 'for '
+            for_flag += 1
+            continue
+        if token == 'IN':
+            python_code += 'in '
+            continue
+        if token == 'NUM' or token == 'BOOL':
+            continue
+        if token == 'SET':
+            if ind == len(tokens) - 1:
+                logger.error(f'Wrong set declaration')
+                return
+            name = tokens[ind + 1]
+            python_code += name + ' = np.array([])\n'
+            continue
+        if token == 'POW':
+            python_code += '** '
+            continue
+        if token == 'DIV':
+            python_code += '// '
+            continue
+        if token == 'MOD':
+            python_code += '% '
+            continue
+        if token == 'T':
+            python_code += 'True '
+            continue
+        if token == 'F':
+            python_code += 'False '
+            continue
+        if token in {'NOT', 'AND', 'OR}'}:
+            python_code += token.lower()+' '
+            continue
+        if token == '|=':
+            if ind == 0:
+                logger.error(f'Wrong {token} ')
+                return
+            name = tokens[ind - 1]
+            python_code += '= ' + name + ' or '
+            continue
```

Comment:
Am I correct in guessing, that `my_set |= another_set` will result in something like `my_set= my_set or another_set`? Int that case, `|=` should not be replaced
If it is used in context with `list`, it doesn't work as lists do not support this operation